src/heap_utils.cc

```diff
@@ -454,6 +455,7 @@ void TriggerHeapSnapshot(const FunctionCallbackInfo<Value>& args) {
 
   if (filename_v->IsUndefined()) {
     DiagnosticFilename name(env, "Heap", "heapsnapshot");
+    // TODO(rafaelgss): resolve(name) and THROW_IF_INSUFFICIENT_PERMISSIONS
```

Comment:
I think in this case you can just check if write to cwd is permitted and the file name does not matter, it's almost impossible for the user to match the exact name generated here from the permission setting anyway.